packages/react-pose/src/components/Transition/children.ts

```diff
@@ -122,13 +123,35 @@ export const handleIncomingChildren = (props: MergeChildrenProps) => {
   }
 };
 
+const VALID_KEY_WARN = '<Transition /> (or <PoseGroup/>) children have to have a valid `key` prop.';
+
 export const makeChildList = (
   children: Array<ReactElement<any>> | ReactElement<any>
 ) => {
   const list: Array<ReactElement<any>> = [];
   Children.forEach(
     children,
-    child => child && list.push(child as ReactElement<any>)
+    child => {
+      if (!child) {
+        return;
+      }
+
+      warning(
+        child.type !== 'string',
+        '<Transition /> (or <PoseGroup/>) children have to be composite components.\n'
+        'Instead it received: ${child.type}.'
+      );
+
+      if (process.env.NODE_ENV !== 'production') {
+        if (list.length === 1) {
```

Comment:
I don't understand when this would catch what the following warning wouldn't?

Reply:
a single child, I assume key-less single child is allowed, or isnt it? if it isnt then ofc this can be removed

Reply:
Yeah all children must have a key otherwise we don't know if a child has been switched out